### ISSUE TITLE
False positive for unused import when using type as string with generic class behind TYPE_CHECKING flag

### ISSUE THREAD

**Issue by last-partizan** [charliermarsh/ruff#4654]
Hello, this happens when using django models with `django-types` and `ForeignKey["SomeModel"]`, but here's minimal example without django:

```python
from typing import TYPE_CHECKING, Generic, TypeVar


if TYPE_CHECKING:
    from pathlib import Path

T = TypeVar("T")


class ForeignKey(Generic[T]):
    ...


class Foo:
    var = ForeignKey["Path"]()
```

```sh
> ruff --version
ruff 0.0.270
> ruff --isolated /tmp/test.py
/tmp/test.py:5:25: F401 [*] `pathlib.Path` imported but unused
Found 1 error.
```

**Comment by charliermarsh** [comment]
Ruff isn't able to do that level of reasoning right now (i.e., to know that `ForeignKey` is a generic, and so `"Path"` should be treated as a forward reference). We _could_ special-case members of `django-types` to support this.

**Comment by last-partizan** [comment]
It would be nice.

If i understand correctly, `ruff` does not know about `django-types` either, so it would be django-specific special case.

How would it looks like?

I think it can be some option like `treat-as-generics = []` with default pointing to django `models.ForeignKey`, and maybe some others.

**Comment by JonathanPlasse** [comment]
You can use the [typing-modules](https://beta.ruff.rs/docs/settings/#typing-modules) setting.

**Comment by last-partizan** [comment]
@JonathanPlasse can you give an example?

In django-types ForeignKey is defined here:

https://github.com/sbdchd/django-types/blob/6e63b611329e5a8fc7b9e04e4a52e8c980734c9b/django-stubs/db/models/fields/related.pyi#L173

**Comment by JonathanPlasse** [comment]
Sorry, I got it wrong, a setting needs to be added to extend the annotated subscripts.

**Comment by JonathanPlasse** [comment]
I am working on adding the setting `extend-annotated-subscripts`.
